app/services/sso/openid_connect.rb

```diff
@@ -42,11 +42,15 @@ def bearer_token_set?
     end
 
     def valid_issuer?
+      logger = Foreman::Logging.logger('app')
+      unless jwt_token.decoded_payload.present?
+        logger.error "Nil JWT received, please check connectivity with the OpenID Provider"
+        false
```

Comment:
I'm assuming this should be 
```suggestion
        return false
```
and then there is no need for the `try`s below since we know the payload isn't `nil`

Reply:
good catch!